sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/EventGridExtensionConfigProvider.cs

```diff
@@ -174,6 +183,50 @@ private async Task<HttpResponseMessage> ProcessEventsAsync(JArray events, string
             return new HttpResponseMessage(HttpStatusCode.Accepted);
         }
 
+        private async Task<FunctionResult> ExecuteWithTracingAsync(string functionName, TriggeredFunctionData triggerData)
+        {
+            using DiagnosticScope scope = _diagnosticScopeFactory.CreateScope(functionName, DiagnosticScope.ActivityKind.Consumer);
+            if (scope.IsEnabled)
+            {
+                if (triggerData.TriggerValue is JArray evntArray)
```

Comment:
I might be missing something but is TriggerValue always a Newtonsoft type? Can it be binary data or a strongly typed model?

Reply:
it's always JArray  and it's converted to POCO or Azure Messaging event type later

https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/EventGridExtensionConfigProvider.cs#L136